src/llm_vm/onsite_llm.py

```diff
@@ -527,3 +485,71 @@ def finetune(self, dataset, optimizer, c_id, small_model_filename=None):
         # optimizer.storage.set_training_in_progress(c_id, False)
         # if old_model is not None:
         #     openai.Model.delete(old_model)
+
+
+class BaseCtransformersLLM(BaseOnsiteLLM):
+    """
+    Base Class for running Ctransformers/GGML models
+
+    Attributes:
+        model_uri (str): Ctransformers uri for LLM
+        model_kwargs (dict): Keyword arguments for loading the LLM
+    
+    Methods:
+        model_loader: Loads specified model from Ctransformers
+        generate: Generates a response from given prompt
+
+    """
+
+    def __init__(self, **model_kwargs):
+        self.__model_uri = None
+        self.__model_file = None
+        self.model = self.model_loader(**model_kwargs)
+
+    @property
+    def model_file(self):
+        return self.__model_file
+
+    @model_file.setter
+    def model_file(self,val):
+        self.__model_file=val 
+
+    @property
+    def model_uri(self):
```

Comment:
We’re doing property for mode uri again when it’s setup the same way in the base class, what’s different ?

Reply:
Why can’t they just use self.actual variable name ? Is there some nuance I don’t know very well?

Reply:
> We’re doing property for mode uri again when it’s setup the same way in the base class, what’s different ?

i had done that initially but ran into errors. later realized that if we don't set up the property here it won't be inherited by the child class